### ISSUE TITLE
Parser got a wrong result : "popq ($rsi)"

### ISSUE THREAD

**Issue by rejohn** [dyninst/dyninst#806]
@hainest ,hi,
    I parser some object file, and I found parser instruction "popq ($rsi)；retq" will get a wrong result:
(pop to a register is right, but pop to a memoryaddress ,the result is wrong):

objdump info:

00000000000000000 <tt>:
	0:	55		push %rbp
	1:	48 89 e5		mov %rsp,%rbp
	4:	68 12 12 00 00 	pushq %0x1212
	9:	8f 06		popq （%rsi）
	b:	c3		retq
	c:	90		nop
	d:	5d		pop %rbp
	e:	c3		retq

And use dyninst to generate CFG info, I see:

	instructions {
	  ea: 9
	  bytes: "\217\006\303\220"
	}

the instructions stream of popq (%rsi)  【8f 06】and ret【c3】 and nop【90】  were parser into  one instrunction "\217\006\303\220" ...

**Comment by hainest** [comment]
@mxz297 @sashanicolas Could one of you look into this?

**Comment by rejohn** [comment]
Hi @mxz297 ,
here is the object file:

[tested_api.zip](https://github.com/dyninst/dyninst/files/5128531/tested_api.zip)

**Comment by mxz297** [comment]
@rejohn Did you post a wrong file? I unzip the tested_api.zip and got a .o file. Objdump shows:

```
tested_api.o:     file format elf64-x86-64


Disassembly of section .text:

0000000000000000 <tested_api>:
   0:   68 12 12 00 00          pushq  $0x1212
   5:   5b                      pop    %rbx
   6:   48 89 1e                mov    %rbx,(%rsi)
   9:   c3                      retq   
```

It does not contain the problematic `popq ($rsi)` instruction.

**Comment by rejohn** [comment]
@mxz297 , I am sorry for my mistake,
here is the object file:
[tested_api.zip](https://github.com/dyninst/dyninst/files/5130436/tested_api.zip)

**Comment by mxz297** [comment]
@rejohn Thanks. I have confirmed that this is an instruction decoder problem. With Capstone decoding backend, Dyninst's output seems correct:

```
0 push %rbp
1 mov %rsp,%rbp
4 push %rbp
5 mov %rsp,%rbp
8 push $0x1212
d pop 0x0(%rsi)
f ret_near (%rsp)
```

Below is the output from objdump:

```
/home/xm13/test-binaries/tested_api.o:     file format elf64-x86-64


Disassembly of section .text:

0000000000000000 <tested_api>:
   0:   55                      push   %rbp
   1:   48 89 e5                mov    %rsp,%rbp
   4:   55                      push   %rbp
   5:   48 89 e5                mov    %rsp,%rbp
   8:   68 12 12 00 00          pushq  $0x1212
   d:   8f 06                   popq   (%rsi)
   f:   c3                      retq   
  10:   90                      nop
  11:   5d                      pop    %rbp
  12:   c3                      retq   
```

You can try my branch following the install direction here:

https://github.com/mxz297/dyninst/blob/layout_opt/branch_build.md

My branch has Capstone backend integrated.

**Comment by rejohn** [comment]
@mxz297 thanks,  Does dyninst have any plans to fix this problem? By the way, will capstore be integrated into the Dyninst ？backbone?

**Comment by mxz297** [comment]
I don't think we plan to fix this decoding problem. Our plan is to integrate Capstone asap.

**Comment by rejohn** [comment]
@mxz297 Thanks，According to the plan, which version will integrate capstone ?

**Comment by mxz297** [comment]
We are still planing the next release, and hopefully the next release will have Capstone integrated. 

You can follow the step at 

https://github.com/mxz297/dyninst/blob/layout_opt/branch_build.md 

to see if that feature branch satisfies all your need, which has capstone integrated. I hope to merge it to dyninst mainstream in the future.
